starters/solidstart-tanstackquery-tailwind-modules/src/components/FetchExample/Greeting.tsx

```diff
@@ -0,0 +1,49 @@
+import { useLocation } from '@solidjs/router';
+import greetingFetcher from './greetingFetcher';
+import { createEffect, createSignal, Match, Switch } from 'solid-js'
+import { createQuery } from '@tanstack/solid-query';
+
+
+const Greeting = () => {
+  const [message, setMessage] = createSignal<string>('');
+  const location = useLocation();
+
+   const query = createQuery(() => [location.query.greeting],  () => greetingFetcher(location.query?.greeting ?? ''));
+
+  createEffect(() => {
+    if (query.isSuccess) {
+      setMessage(query.data)
+    }
+  })
+
+  return (
+    <div class="w-1/4 mx-auto">
+      <Switch>
+        <Match when={query.isLoading}>
+           <div class="grow animate-pulse rounded-md bg-gray-200 text-left">
+              Loading...
+            </div>
+        </Match>
+        <Match when={query.isError}>
+           <div
+          class="grow rounded border border-solid border-red-300 bg-red-100 p-4 text-center text-red-500"
+          role="error-message"
```

Comment:
This `role` attribute has a typescript warning, I think we can remove this since we never use it and it's not necessary.